scarlet/source.py

```diff
@@ -395,6 +396,101 @@ def __init__(self, frame, sky_coord, observations):
     def _psf_wrapper(self, *parameters):
         return self.frame.psf.__call__(*parameters, bbox=self.bbox)[0]
 
+class StarletSource(FunctionComponent):
+    """Source intialized with starlet coefficients.
+
+    Sources are initialized with the SED of the center pixel,
+    and the morphologies are initialised as ExtendedSources
+    and transformed into starlet coefficients.
+    """
+
+    def __init__(
+        self,
+        frame,
+        sky_coord,
+        observations,
+        noise,
```

Comment:
I don't think it is necessary to force the user to compute this. Simply compute the transform, compute the MAD of the coefficients and you're good. cf `ExtendedSource`, where we also make the noise threshold computations ourself.

Reply:
I thought I had a good reason to do that but now it eludes me.

Reply:
Oh, I know, that's in case the noise is not gaussian. Here you could imagine giving a full rms map.

Reply:
If the noise is not Gaussian, we're in deeper trouble (the log likelihood isn't right then). If it's spatially variable it would be stored in `observation.weights`, it is passed to the source constructor already. It's part of the job of `...Source` to set up the parameters properly, so I don't see a good reason to make the user perform very implementation-specific computations.